protocols/kad/src/handler.rs

```diff
@@ -786,6 +786,8 @@ where
                 }
             }
         }
+
+        debug_assert!(false, "Cannot find inbound substream for {request_id:?}")
```

Comment:
Just to test my intuition, given that we never poll an underlying inbound stream in `InboundSubstreamState::WaitingUser` state it won't close in the meantime and thus it won't be removed before we call `answer_pending_request`, correct?

Reply:
Yes.